### PR TITLE
6683: refactor case deadline storage and retrieve by date range

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

File: shared/src/persistence/dynamo/caseDeadlines/createCaseDeadlineCatalogRecord.js [hunk 1]
```diff
@@ -10,12 +10,14 @@ const { put } = require('../../dynamodbClientService');
...
+  deadlineDate,
 }) => {
   await put({
     Item: {
       caseDeadlineId,
-      pk: 'case-deadline-catalog',
-      sk: `case-deadline|${caseDeadlineId}`,
+      gsi1pk: 'case-deadline-catalog',
+      pk: deadlineDate,
+      sk: `case-deadline-catalog|${caseDeadlineId}`,
```
Comment (matthopson) [inline shared/src/persistence/dynamo/caseDeadlines/createCaseDeadlineCatalogRecord.js:20]: Do you think it makes more sense for the `deadlineDate` to be on the sort key? Or am I being pedantic?

Comment (rachaelparris) [reply shared/src/persistence/dynamo/caseDeadlines/createCaseDeadlineCatalogRecord.js:20]: It surely does! This is literally the only way I could set this up for the queries to all work in the way we needed them to. I'm making use of the global secondary index's use of pk as the range_key.

Comment (rachaelparris) [reply shared/src/persistence/dynamo/caseDeadlines/createCaseDeadlineCatalogRecord.js:20]: I agree that it's unconventional and maybe I should add a comment stating why it is this way (although test failures would be a clue).

File: web-client/src/presenter/actions/CaseDeadline/setDefaultCaseDeadlinesReportDatesAction.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { state } from 'cerebral';
+
+/**
+ * sets default start and end date for case deadlines report to today's date
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @returns {object} the case deadlines
```
Comment (matthopson) [inline web-client/src/presenter/actions/CaseDeadline/setDefaultCaseDeadlinesReportDatesAction.js:9]: missing store

File: shared/src/business/useCases/getCaseDeadlinesInteractor.js [hunk 3]
```diff
@@ -8,13 +8,17 @@ const { pick } = require('lodash');
 const { UnauthorizedError } = require('../../errors/errors');
 
 /**
- * getAllCaseDeadlinesInteractor
+ * getCaseDeadlinesInteractor
  *
  * @param {object} providers the providers object
  * @param {object} providers.applicationContext the application context
  * @returns {Promise} the promise of the getCaseDeadlines call
```
Comment (matthopson) [inline shared/src/business/useCases/getCaseDeadlinesInteractor.js]: Update jsdocs

File: shared/src/proxies/caseDeadline/getCaseDeadlinesProxy.js [hunk 4]
```diff
@@ -1,15 +1,19 @@
 const { get } = require('../requests');
 
 /**
- * getAllCaseDeadlinesInteractorProxy
+ * getCaseDeadlinesInteractorProxy
  *
  * @param {object} providers the providers object
  * @param {object} providers.applicationContext the application context
  * @returns {Promise<*>} the promise of the api call
```
Comment (matthopson) [inline shared/src/proxies/caseDeadline/getCaseDeadlinesProxy.js:10]: jsdocs

File: web-client/src/presenter/actions/CaseDeadline/getCaseDeadlinesAction.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { state } from 'cerebral';
+
+/**
+ * get case deadlines between start and end date
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @returns {object} the case deadlines
```
Comment (matthopson) [inline web-client/src/presenter/actions/CaseDeadline/getCaseDeadlinesAction.js:9]: missing get

**Review by matthopson — APPROVED** [review]
Just some jsdocs nits

**Review by matthopson — APPROVED** [review]

**Pull request merged**